src/tests/time-base-api-test/time-base-api-test.c

```diff
@@ -0,0 +1,278 @@
+/*
+ *      Copyright (c) 2020, United States government as represented by the
+ *      administrator of the National Aeronautics Space Administration.
+ *      All rights reserved. This software was created at NASA Goddard
+ *      Space Flight Center pursuant to government contracts.
+ *
+ *      This is governed by the NASA Open Source Agreement and may be used,
+ *      distributed and modified only according to the terms of that agreement.
+ */
+
+/*
+ * Filename: time-base-api-test.c
+ *
+ * Purpose: This file contains functional tests for "osapi-timebase"
+ *
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "common_types.h"
+#include "osapi.h"
+#include "utassert.h"
+#include "uttest.h"
+#include "utbsp.h"
+
+static uint32 TimerSyncCount = 0;
+static uint32 TimerSyncRetVal = 0;
+
+static uint32 UT_TimerSync(uint32 timer_id)
+{
+    ++TimerSyncCount;
+    return TimerSyncRetVal;
+}
+
+
+/* *************************************** MAIN ************************************** */
+
+void TestTimeBaseApi(void)
+{
+    int32 expected;
+    int32 actual; 
+    int32 TimeBaseNum;
+    uint32 freerun;
+    uint32 objid;
+    uint32 time_base_id; 
+    uint32 time_base_id2; 
+    char maxTimeBase[12];
+    char overMaxTimeBase[12];
+    OS_timebase_prop_t timebase_prop;
+
+    /*
+     * Test Case For:
+     * int32 OS_TimeBaseCreate(uint32 *timer_id, const char *timebase_name, OS_TimerSync_t external_sync)
+     */
+
+    /* Test for nominal inputs */
+    expected = OS_SUCCESS;
+
+    actual= OS_TimeBaseCreate(&time_base_id, "TimeBase", 0);
+    UtAssert_True(actual == expected, "OS_TimeBaseCreate() (%ld) == OS_SUCCESS", (long)actual);
+
+    actual = OS_TimeBaseCreate(&time_base_id2, "TimeBase2", NULL);
+    UtAssert_True(actual == expected, "OS_TimeBaseCreate() (%ld) == OS_SUCCESS", (long)actual);
+
+    actual = OS_TimeBaseCreate(&time_base_id, "TimeBase3", UT_TimerSync);
+    UtAssert_True(actual == expected, "OS_TimeBaseCreate() (%ld) == OS_SUCCESS", (long)actual);
+
+    /* Test for nominal, max/min cases */
+    objid = 0xFFFFFFFF;
+    actual = OS_TimeBaseCreate(&objid, "TimeBase4", 0);
+    UtAssert_True(actual == expected, "OS_TimeBaseCreate() (%ld) == OS_SUCCESS", (long)actual);
+
+    /* 
+     * Note: OS_MAX_TIMEBASES = 5, so no more than 5 TimeBase IDs are allowed 
+     * Checking for OS_MAX_TIMEBASES:
+     */
+    TimeBaseNum = OS_MAX_TIMEBASES;
+    snprintf(maxTimeBase, 12, "TimeBase%d", TimeBaseNum);
+    objid = 0x00000000;
+    actual = OS_TimeBaseCreate(&objid, "maxTimeBase", 0);
+    UtAssert_True(actual == expected, "OS_TimeBaseCreate() (%ld) == OS_SUCCESS", (long)actual);
+
```

Comment:
The OS_MAX_TIMEBASES is a configurable parameter.  Could we turn this into a loop based on OS_MAX_TIMEBASES so that this test case still works if the user changes this value?

Reply:
I read in the OS_MAX_TIMEBASES, so shouldn't it automatically always read in the max value if the user reconfigured it it before this point?

Reply:
The idea is to confirm that one can create entities all the way up to OS_MAX_TIMEBASES (i.e. all OS_SUCCESS), then attempt to create one more which should return an error.  The point at which this happens is user configurable so this would have to be converted into a loop.  This is currently hardcoding 5 timebase objects, so it will fail with OS_MAX_TIMEBASES set to anything other than 5.